api/v1alpha1/eventing_types.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EventingSpec defines the desired state of Eventing
+type EventingSpec struct {
+	// Backends defines the list of eventing backends to provision.
+	Backends []Backend `json:"backends"`
+
+	// Publisher defines the configurations for eventing-publisher-proxy.
+	// +optional
+	Publisher `json:"publisher,omitempty"`
+
+	// Logging defines the log level for eventing-manager.
+	// +optional
+	Logging `json:"logging,omitempty"`
+
+	// Annotations allows to add annotations to resources.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Labels allows to add Labels to resources.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// EventingStatus defines the observed state of Eventing
+type EventingStatus struct {
+	State      string             `json:"state"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Eventing is the Schema for the eventings API
```

Comment:
this way it becomes clearer that the decorators belong to `Eventing`.

```suggestion
// Eventing is the Schema for the eventing API.
//+kubebuilder:object:root=true
//+kubebuilder:subresource:status
```